### PR TITLE
Use vscode built in support for block cursors

### PR DESCRIPTION
**Pull request by Paxxi**
Added some simple tests for this and moved it out of the hot path as it's something that only happens during mode changes.

### CODE REVIEW

**Comment by jpoon** [comment]
@Paxxi sweet! there's a minor tslint error that's making the build fail

> [16:16:39] [gulp-tslint] error mode/modeHandler.ts[195, 13]: unused variable: '_caretDecoration'

**Comment by Paxxi** [comment]
Ahh right, forgot to run tslint, will fix tomorrow

**Comment by Paxxi** [comment]
Fixed the tslint error and rewrote it a bit to make it a bit shorter

File: test/mode/modeHandler.test.ts [hunk 1]
```diff
@@ -32,4 +33,14 @@ suite("Mode Handler", () => {
...
+
+    test("Uses correct cursor style depending on mode", async () => {
+        const modeHandler = new ModeHandler();
+
+        modeHandler.setCurrentModeByName(ModeName.Normal);
+        assert.equal(vscode.window.activeTextEditor.options.cursorStyle, vscode.TextEditorCursorStyle.Block);
+
+        modeHandler.setCurrentModeByName(ModeName.Insert);
+        assert.equal(vscode.window.activeTextEditor.options.cursorStyle, vscode.TextEditorCursorStyle.Line);
+    });
```
Comment (jpoon) [inline test/mode/modeHandler.test.ts]: Can you add another case here for Visual mode?

**Comment by Paxxi** [comment]
Ok added the test case.

Note for the future, I can be sloppy sometimes in PR's so don't hesitate to "nitpick", I expect it :smile:

**Comment by jpoon** [comment]
@Paxxi one more tiny thing, can you merge the latest changes from `master` to this branch?

**Comment by Paxxi** [comment]
done

**Comment by jpoon** [comment]
Fixes https://github.com/VSCodeVim/Vim/issues/132! Thanks @Paxxi!

**Comment by johnfn** [comment]
@Paxxi unfortunately this breaks visual line mode :( In visual line mode, the block cursor is often in the middle of the linewise selection (Vim will always select the entire line the cursor is in, no matter if it's at the beginning or the end).

Can you find any way to fix this? Otherwise we'll have to revert, which would be a shame because this cursor is really nice.

**Comment by Paxxi** [comment]
I see the issue, will investigate and see how we can solve it

**Comment by Paxxi** [comment]
Not sure if I will have time to investigate in the next few days so maybe revert this in case you want to do a release and we can give it another try once I figure it out

**Pull request merged**